redos/7/Dockerfile

```diff
@@ -0,0 +1,57 @@
+FROM centos:7 as prepare
+MAINTAINER Sergei Vorontsov "piligrim@rootnix.net"
+
+ENV REPO_URL=http://mirror.yandex.ru/redos/7.3/x86_64/os/ \
+    CHROOT_DIR="/chroot"
+
+RUN mkdir -p $CHROOT_DIR
+
+RUN for rpm in `curl -s $REPO_URL | grep 'redos-release' | sed 's/<.*>\(.*\)<\/.*>/\1/' | awk '{print $1}'`; do if [[ $rpm =~ redos-release-[0-9].[0-9]-([0-9]|[0-9][0-9]).el7.noarch.rpm ]]; then curl -o $CHROOT_DIR/redos-release.rpm $REPO_URL$rpm; fi; done
+
+#setup base system
+RUN rpm --root=$CHROOT_DIR --rebuilddb
+RUN rpm --root=$CHROOT_DIR --nodeps --quiet -i $CHROOT_DIR/redos-release.rpm
+RUN cp $CHROOT_DIR/etc/pki/rpm-gpg/RPM-GPG-KEY-RED-SOFT /etc/pki/rpm-gpg/RPM-GPG-KEY-RED-SOFT
+
+RUN yum -y --installroot=$CHROOT_DIR update
+
+RUN yum -y --installroot=$CHROOT_DIR install -y \
+    yum \
+    rpm-devel \
+    rpm-devel \
+    devtoolset \
+	rpmdevtools \
+    rpm-build \
+    gcc \
+    gcc-c++ \
+    cmake \
+    make \
+    ccache \
+    git \
+    sudo \
+    yum-utils \
+    zypper \
+	gettext-devel \
+    sed \
+    readline-devel \
+    openssl-devel \
+    libicu-devel \
+    libffi-devel \
+    python3-pip \
+    python3-devel \
+    python3-zope-interface \
+    python3-greenlet
+
+RUN sed -i.bak -n -e '/^Defaults.*requiretty/ { s/^/# /;};/^%wheel.*ALL$/ { s/^/# / ;} ;/^#.*wheel.*NOPASSWD/ { s/^#[ ]*//;};p' $CHROOT_DIR/etc/sudoers
+
+#Clear cache
+RUN rm -rf $CHROOT_DIR/var/cache/yum
+
+FROM scratch
+
+WORKDIR /
+
+COPY --from=prepare /chroot/. /
+
+#Rpm packages in Redos repos are broken, so we need to install them manually
+RUN pip3 install gevent
```

Comment:
It would be convenient to have `CMD ["/bin/bash"]` at the end to run a container without explicit `/bin/bash`.